admin/src/provideAppState.js

```diff
@@ -10,12 +10,19 @@ export const state = {
         environmentName: 'development',
     }),
     effects: {
-        setConfig: softUpdate((state, { origin, projectId, token, secret }) => ({
-            origin,
-            projectId,
-            token,
-            secret,
-        })),
+        setConfig: softUpdate((state, { origin, projectId, token, secret }) => {
+            sessionStorage.setItem('comfy.origin', origin);
```

Comment:
do not do this in the function that update the state
do
```js
 (effects, { origin, projectId, token, secret }) => {
    sessionStorage.setItem('comfy.projectId', projectId);
    sessionStorage.setItem('comfy.token', token);
    sessionStorage.setItem('comfy.secret', secret);
    return state => ({
                ...state
                origin,
                projectId,
                token,
                secret,
            });
```

to separate sideEffect from state update